snapshots/types/snapshotter.go

```diff
@@ -1,16 +1,33 @@
 package types
 
-import "io"
+import (
+	protoio "github.com/gogo/protobuf/io"
+)
 
 // Snapshotter is something that can create and restore snapshots, consisting of streamed binary
 // chunks - all of which must be read from the channel and closed. If an unsupported format is
 // given, it must return ErrUnknownFormat (possibly wrapped with fmt.Errorf).
 type Snapshotter interface {
-	// Snapshot creates a state snapshot, returning a channel of snapshot chunk readers.
-	Snapshot(height uint64, format uint32) (<-chan io.ReadCloser, error)
+	// Snapshot writes snapshot items into the protobuf writer.
+	Snapshot(height uint64, protoWriter protoio.Writer) error
```

Comment:
are we okay accepting this api breakage for 0.45? @amaurym

Reply:
I think the `Snapshotter` interface is only used externally before.
And if the API breakage is a concern, we can create a new name, and leave the previous `Snapshotter` interface untouched (and not used).

Reply:
ah yes its only internal, then its fine from my side